Error with Cornac

<!--
BEFORE POSTING YOUR ISSUE:
- These comments won't show up when you submit the issue.
- Please use the sections below to provide information about the issue.
- Be specific: Add as much detail as possible.
-->

## Issue Overview
<!-- A brief overview of the issue --->

## Describe your environment
<!-- Provide details about your environment: what editor, browser, and other software you are using and any other specifics to your setup -->
Python 3.7

## Steps to Reproduce
<!-- Provide an unambiguous set of steps to reproduce this bug. Include code to reproduce, if relevant. Include a live link if available. -->
```
python -m pip install -q --upgrade pip setuptools wheel
  pip install -q --no-use-pep517 lightfm
  pip install -q -r requirements.txt
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/LinkedInLearning/recommendation-systems-a-practical-introduction-2703578/actions/runs/8608761794/job/23591646429#step:4:3).7.17/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.7.17/x6[4](https://github.com/LinkedInLearning/recommendation-systems-a-practical-introduction-2703578/actions/runs/8608761794/job/23591646429#step:4:4)/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x[6](https://github.com/LinkedInLearning/recommendation-systems-a-practical-introduction-2703578/actions/runs/8608761794/job/23591646429#step:4:6)4
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.[7](https://github.com/LinkedInLearning/recommendation-systems-a-practical-introduction-2703578/actions/runs/8608761794/job/23591646429#step:4:7).17/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.7.17/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.17/x64/lib
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [2 lines of output]
      We need some dependencies to build Cornac.
      Run: pip3 install Cython numpy scipy
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: Process completed with exit code 1.
```

## Expected Behavior
<!-- What behavior did you expect? -->

## Current Behavior
<!-- What happened instead of the expected behavior? Describe the difference. -->

## Possible Solution
<!-- Optional: Do you have a fix or a suggestion on how to fix the issue? -->

## Screenshots / Video
<!-- Optional: Add any screenshots or video of the issue if available. -->

## Related Issues
<!-- List related issues -->

--- a/.github/workflows/nightly_builds.yml
+++ b/.github/workflows/nightly_builds.yml
@@ -47,7 +47,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install -q --upgrade pip setuptools wheel
-        pip install -q --no-use-pep517 lightfm
+        pip install -q numpy "Cython<4" "scipy<1.11.0"
         pip install -q -r requirements.txt
 
     - name: Python version and dependency list

--- a/.github/workflows/pr_gate.yml
+++ b/.github/workflows/pr_gate.yml
@@ -43,6 +43,7 @@ jobs:
       run: |
         python -m pip install -q --upgrade pip setuptools wheel
         pip install -q --no-use-pep517 lightfm
+        pip install -q numpy "Cython<4" "scipy<1.11.0"
         pip install -q -r requirements.txt
 
     - name: Python version and dependency list

--- a/.github/workflows/pr_gate.yml
+++ b/.github/workflows/pr_gate.yml
@@ -42,7 +42,6 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install -q --upgrade pip setuptools wheel
-        pip install -q --no-use-pep517 lightfm
         pip install -q numpy "Cython<4" "scipy<1.11.0"
         pip install -q -r requirements.txt
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,2 @@
-pandera<0.18.0
-Cython<4
-numpy>=1.19
-scipy<1.11.0
 pytest>=3.6.4
-git+https://github.com/microsoft/recommenders.git@main
+recommenders==1.1.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,3 @@
+pandera<0.18.0
 pytest>=3.6.4
 recommenders==1.1.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,3 +1,6 @@
 pandera<0.18.0
-git+https://github.com/microsoft/recommenders.git@main
+Cython<4
+numpy>=1.19
+scipy<1.11.0
 pytest>=3.6.4
+git+https://github.com/microsoft/recommenders.git@main

